### PR TITLE
bpo-36345: Include the code from Tools/scripts/serve.py for the wsgiref-base web server example

### PR DESCRIPTION
**Pull request by matrixise**
In this PR, I include the code from `Tools/scripts/serve.py` file but I don't touch the code of the script, just keep it intact because it's not the role of this PR.

https://bugs.python.org/issue36345

### CODE REVIEW

**Comment by vstinner** [comment]
make check suspicious html SPHINXOPTS="-q -W -j4" failed with the following error on Travis CI:

```
Warning, treated as error:
[whatsnew/changelog:28] ".. literalinclude:" found in ".. literalinclude::"
```

I'm not sure why literalinclude is seen as an error?

@JulienPalard: Any idea?

**Comment by matrixise** [comment]
@JulienPalard @vstinner ?

I have made the requested changes; please review again

File: Doc/library/wsgiref.rst
Comment (vstinner) [inline Doc/library/wsgiref.rst]: Can you explain what the server does? It seems like it serves the directory passed as the first argument on the command line, and accept an optional port number as the second parameter. Proposition: 

"Example of a WSGI application serving the directory passed on the command line:"

By the way, would it make sense to modify the example of use the current directory by default, rather than require to pass a directory?

```
    path = sys.argv[1] if len(sys.argv) > 1 else os.getcwd()
```

with:

"Example of a WSGI application serving the current directory, accept optional directory and port number on the command line:"

**Comment by matrixise** [comment]
>By the way, would it make sense to modify the example of use the
>current directory by default, rather than require to pass a directory?
>
>```
>    path = sys.argv[1] if len(sys.argv) > 1 else os.getcwd()
>```
>
>with:
>
>"Example of a WSGI application serving the current directory, accept
>optional directory and port number on the command line:"

I am fine with the optional directory, and because it's an example, it's
not a problem, we don't break the backward compatibility.

+1

**Comment by matrixise** [comment]
I have made the requested changes; please review again

File: Tools/scripts/serve.py
Comment (vstinner) [inline Tools/scripts/serve.py:31]: When I press CTRL+C, I get a ResourceWarning. Please add httpd.server_close() after print("\b\bShutting down."). By the way, I'm not sure that "\b" is supported on Windows. I suggest to remove ""\b\b" to make the script more portable.

File: Doc/library/wsgiref.rst
Comment (vstinner) [inline Doc/library/wsgiref.rst]:
```suggestion
directory and port number (default: 8000) on the command line:
```

**Review by vstinner — COMMENTED** [review]
When I tested the example, it wasn't obvious to me how to access the server.

Maybe the script should display somehow the URL http://localhost:8000/?

**Comment by matrixise** [comment]
>When I tested the example, it wasn't obvious to me how to access the server.
Yep, it's the default output of `Tools/scripts/serve.py`

>Maybe the script should display somehow the URL http://localhost:8000/?
Like the output of python -m http.server -d DIRECTORY?

I propose to submit an other bpo issue/pull request where we suggest to
change the default output of the `Tools/scripts/serve.py`

What do you think?

>
>> @@ -25,7 +25,7 @@ def app(environ, respond):
>         return [b'not found']
>
> if __name__ == '__main__':
>-    path = sys.argv[1]
>+    path = sys.argv[1] if len(sys.argv) > 1 else os.getcwd()
>     port = int(sys.argv[2]) if len(sys.argv) > 2 else 8000
>     httpd = simple_server.make_server('', port, app)
>     print("Serving {} on port {}, control-C to stop".format(path, port))
>
>When I press CTRL+C, I get a ResourceWarning. Please add
>httpd.server_close() after print("\b\bShutting down."). By the way, I'm
>not sure that "\b" is supported on Windows. I suggest to remove "\b\b"
>to make the script more portable.
:+1:

File: Tools/scripts/serve.py
Comment (matrixise) [reply Tools/scripts/serve.py:31]: Fixed

**Review by vstinner — APPROVED** [review]
LGTM. You can merge it once you will be able to merge changes ;-)

**Review by berkerpeksag — COMMENTED** [review]
Two things:

1. Only big documentation changes require a NEWS entry.
2. We only attribute contributors in NEWS entries. It's OK to add your name to `Doc/whatsnew/X.Y.rst` even if you're a core developer, though.

**Comment by vstinner** [comment]
> Only big documentation changes require a NEWS entry.

I'm wasn't sure on that point. In case of doubt, I said nothing :-)

> We only attribute contributors in NEWS entries. It's OK to add your name to Doc/whatsnew/X.Y.rst even if you're a core developer, though.

Well, Stéphane wrote this PR before he was a core dev :-)

**Comment by matrixise** [comment]
@berkerpeksag Thank you for your advice, which I will use next time.

**Comment by berkerpeksag** [comment]
> Well, Stéphane wrote this PR before he was a core dev :-)

I'm not sure how that is an argument here? It could be easily noticed and fixed *before* merging the PR.

@matrixise welcome!

**Pull request merged**
